File: Tutorials/VB/Tutorials/Tutorial16.vb
Imports Excel = NetOffice.ExcelApi
Imports NetOffice.ExcelApi.ModulesLegacy.ApplicationModule

Public Class Tutorial16
    Implements ITutorial

    Dim _hostApplication As IHost

    Public Sub Run() Implements TutorialsBase.ITutorial.Run

        ' this example demonstrate the global helper module(static class)
        ' the module is a vba compatibility workarround and contains static methods and properties from the coresponding Application class.

        ' start excel and add a new workbook
        Dim application As New Excel.ApplicationClass()
        application.Visible = False
        application.DisplayAlerts = False
        application.Workbooks.Add()

        ' ApplicationModule contains the well known globals and is located in NetOffice.$XXXApi.ModulesLegacy
        ' This helpful to bring code from VBA to NetOffice
        ActiveCell.Value = "ActiveCellValue"

        ' close and dispose excel
        application.Quit()
        application.Dispose()

        _hostApplication.ShowFinishDialog()

    End Sub

    Public ReadOnly Property Caption As String Implements TutorialsBase.ITutorial.Caption
        Get
            Return "Tutorial16"
        End Get
    End Property

    Public ReadOnly Property Description As String Implements TutorialsBase.ITutorial.Description
        Get
            Return "Globals in NetOffice"
        End Get
    End Property

    Public Sub Connect(ByVal hostApplication As TutorialsBase.IHost) Implements TutorialsBase.ITutorial.Connect

        _hostApplication = hostApplication

    End Sub

    Public Sub Disconnect() Implements TutorialsBase.ITutorial.Disconnect

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements TutorialsBase.ITutorial.Panel
        Get
            Return Nothing
        End Get
    End Property

    Public ReadOnly Property Uri As String Implements TutorialsBase.ITutorial.Uri
        Get
            Return FormMain.DocumentationBase & "Tutorial16_EN_VB.html"
        End Get
    End Property

End Class

File: Tutorials/C#/Tutorials/Tutorial16.cs
using System;
using System.Windows.Forms;
using TutorialsBase;
using Excel = NetOffice.ExcelApi;
using static NetOffice.ExcelApi.ModulesLegacy.ApplicationModule;

namespace TutorialsCS4
{
    public class Tutorial16 : ITutorial
    {
        public void Run()
        {
            // this example demonstrate the global helper module(static class)
            // the module is a vba compatibility workarround and contains static methods and properties from the coresponding Application class.

            // start excel and add a new workbook
            Excel.Application application = new Excel.ApplicationClass();
            application.Visible = false;
            application.DisplayAlerts = false;
            application.Workbooks.Add();

            // ApplicationModule contains the well known globals and its located in NetOffice.$XXXApi.ModulesLegacy
            // This helpful to bring code from VBA to NetOffice
            ActiveCell.Value = "ActiveCellValue";

            // quit and dispose excel
            application.Quit();
            application.Dispose();

            HostApplication.ShowFinishDialog();
        }

        public void Connect(IHost hostApplication)
        {
            HostApplication = hostApplication;
        }

        public void Disconnect()
        {

        }

        public string Uri
        {
            get { return Program.DocumentationBase + "Tutorial16_EN_CS.html"; }
        }

        public string Caption
        {
            get { return "Tutorial16"; }
        }

        public string Description
        {
            get { return "Globals in NetOffice"; }
        }

        public UserControl Panel
        {
            get { return null; }
        }

        internal IHost HostApplication { get; private set; }
    }
}
